### PR TITLE
Update Demos & Docs

### PR DESCRIPTION
**Pull request by jacobangel**
1. Moves demos to a `presets` folder.
2. Creates a `simple` demo, that shows the minimum viable implementation.
3. Adds some readme info.
4. Adds webpack dev server to run the demos from a common demo root.

### CODE REVIEW

File: webpack.demos.config.js [hunk 1]
```diff
@@ -23,5 +26,11 @@ module.exports = {
   },
   eslint: {
     configFile: '.eslintrc'
+  },
+  devServer: {
```
Comment (maisano) [inline webpack.demos.config.js]: hm, we need to shuffle some stuff around if we're to add `devServer`. when running a clean `npm run start:demos`, the outputs from each demo are off by one directory–they're requesting `/[name]/index.js` but the dev server is serving it at `/demos/[name]/index.js`.

**Comment by maisano** [comment]
this is great–it should definitely help clear up some of the confusion that folks tend to have when trying to use this library. just a minor issue with the webpack config to clear up :)

thanks!

File: webpack.demos.config.js (hunk 1 above)
Comment (jacobangel) [inline webpack.demos.config.js]: should be fixed now!

**Comment by maisano** [comment]
lgtm, thanks @jacobangel

**Pull request merged**